subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -58,6 +58,31 @@ Usage of invalid Java toolchain specifications is no longer supported.
 Related build errors can be avoided by making sure that language version is set on all toolchain specifications.
 See <<toolchains#sec:configuring_toolchain_specifications,user manual>> for more information.
 
+==== Changing test framework after setting test framework options is now an error
+
+When configuring the built-in test task for Java/Groovy/Scala projects, Gradle no longer allows you to
+change the test framework used by the `Test` task after configuring options. This was deprecated and silently
+discarded configuration in some cases.
```

Comment:
```suggestion
change the test framework used by the `Test` task after configuring options. This was deprecated since it silently
discarded configuration in some cases.
```
Suggestion: This doesn't seem like a natural "and" usage to me. I *think* "silently discarded configuration" is actually the reason this feature was deprecated... so we should make that obvious!